src/postprocessors.rs

```diff
@@ -51,6 +52,70 @@ fn filter_by_tags_(
     }
 }
 
+/// This postprocessor removes all Obsidian comments from a file excluding codeblocks. Enabling this
+/// prohibits comments from being exported but leaves them untouched in the original files
+pub fn remove_obsidian_comments(
+    _context: &mut Context,
+    events: &mut MarkdownEvents,
+) -> PostprocessorResult {
+    let mut output = Vec::with_capacity(events.len());
+    let mut inside_comment = false;
+    let mut inside_codeblock = false;
+
+    for event in &mut *events {
+        output.push(event.to_owned());
+
+        match event {
+            Event::Text(ref text) => {
+                if !text.contains("%%") {
+                    if inside_comment {
+                        output.pop(); //Inside block comment so remove
+                    }
+                    continue;
+                } else if inside_codeblock {
+                    continue; //Skip anything inside codeblocks
+                }
+
+                output.pop();
+
+                if inside_comment {
+                    inside_comment = false;
+                    continue;
+                }
+
+                if !text.eq(&CowStr::from("%%")) {
+                    let re = Regex::new(r"%%.*?%%").unwrap();
```

Comment:
Can this be implemented without having to use a regex?

If it does need to be done via a regex, the construction should be wrapped in something like a [LazyCell](https://doc.rust-lang.org/std/cell/struct.LazyCell.html) to only compile it once instead of for each iteration of the loop, as regex compilation is computationally quite heavy.